src/Messages/Messages.php

```diff
@@ -0,0 +1,281 @@
+                $messages->next();
+            }
+        }
+    }
+
+    /**
+     * Returns the number of messages in the list
+     *
+     * @return int
+     */
+    public function count(): int
+    {
+        return count($this->messages);
+    }
+
+    /**
+     * Returns the current message in the iterator
+     *
+     * @return MessageInterface
+     */
+    public function current(): MessageInterface
+    {
+        return $this->messages[$this->position];
+    }
+
+    /**
+     * Filters the message collection by field name
+     *
+     * @param string $fieldName
+     * @return array
+     */
+    public function filter(string $fieldName): array
+    {
+        $filtered = [];
+
+        if (is_array($this->messages)) {
+            /**
+             * A collection of messages is iterated and appended one-by-one to
+             * the current list
+             */
+            foreach ($this->messages as $message) {
+                if (
+                    method_exists($message, 'getField')
+                    && $fieldName === $message->getField()
+                ) {
+                    $filtered[] = $message;
+                }
+            }
+        }
+
+        return $filtered;
+    }
+
+    /**
+     * Moves the internal iteration pointer to the next position
+     */
+    public function next(): void
+    {
+        $this->position++;
+    }
+
+    /**
+     * Returns the current position/key in the iterator
+     *
+     * @return int
+     */
+    public function key(): int
+    {
+        return $this->position;
+    }
+
+    /**
+     * Check if the current message in the iterator is valid
+     *
+     * @return bool
+     */
+    public function valid(): bool
+    {
+        return isset($this->messages[$this->position]);
+    }
+
+    /**
+     * Rewinds the internal iterator
+     */
+    public function rewind(): void
+    {
+        $this->position = 0;
+    }
+
+    /**
+     * Checks if an index exists
+     *
+     *```php
+     * var_dump(
+     *     isset($message["database"])
+     * );
+     *```
+     *
+     * @param int $offset
+     * @return bool
+     */
+    public function offsetExists($offset): bool
+    {
+        return isset($this->messages[$offset]);
+    }
+
+    /**
+     * Gets an attribute a message using the array syntax
+     *
+     *```php
+     * print_r(
+     *     $messages[0]
+     * );
+     *```
+     *
+     * @param mixed $offset
+     * @return mixed|null
+     */
+    public function offsetGet($offset)
+    {
+        return $this->messages[$offset] ?? null;
+    }
+
+    /**
+     * Sets an attribute using the array-syntax
+     *
+     *```php
+     * $messages[0] = new \Phalcon\Messages\Message("This is a message");
+     *```
+     *
+     * @param mixed $offset
+     * @param Message $message
+     * @throws Exception
+     */
+    public function offsetSet($offset, $message): void
+    {
+        if (!is_object($message)) {
```

Comment:
Should this be an `is_object` check or a check for type `MessageInterface`?

Reply:
MessageInterface, I didn't dare to change the existing code